adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantContextConverter.scala

```diff
@@ -300,45 +270,368 @@ private[adam] class VariantContextConverter(
+    vb: Variant.Builder): Variant.Builder = {
+
+    splitIds(vc).fold(vb)(vb.setNames(_))
+  }
+
+  private[converters] def formatFilters(
+    vc: HtsjdkVariantContext,
+    vb: Variant.Builder): Variant.Builder = {
+
+    vb.setFiltersApplied(vc.filtersWereApplied)
     if (vc.filtersWereApplied) {
-      builder.setFiltersPassed(!vc.isFiltered)
+      vb.setFiltersPassed(!vc.isFiltered)
     }
     if (vc.isFiltered) {
-      builder.setFiltersFailed(new java.util.ArrayList(vc.getFilters))
+      vb.setFiltersFailed(new java.util.ArrayList(vc.getFilters));
     }
-    builder.build
+    vb
   }
 
-  /**
-   * Populates a variant annotation from an htsjdk variant context.
-   *
-   * @param variant Avro variant representation for the site.
-   * @param vc htsjdk representation of the VCF line.
-   * @return Returns the Avro representation of the variant annotations at this site.
-   */
-  private def extractVariantAnnotation(variant: Variant,
-                                       vc: HtsjdkVariantContext): VariantAnnotation = {
-    ???
+  private val variantFormatFns: Iterable[(HtsjdkVariantContext, Variant.Builder) => Variant.Builder] = Iterable(
+    formatNames(_, _),
+    formatFilters(_, _)
+  )
+
+  // variant --> htsjdk extract functions
+
+  private[converters] def extractNames(
+    v: Variant,
+    vcb: VariantContextBuilder): VariantContextBuilder = {
+
+    joinNames(v).fold(vcb.noID())(vcb.id(_))
+  }
+
+  private[converters] def extractFilters(
+    v: Variant,
+    vcb: VariantContextBuilder): VariantContextBuilder = {
+
+    Option(v.getFiltersApplied)
+      .filter(ft => ft)
+      .map(applied => {
+        Option(v.getFiltersPassed).map(passed => {
+          if (passed) {
+            vcb.passFilters
+          } else {
+            val failedFilters = v.getFiltersFailed
+            require(failedFilters.nonEmpty,
+              "Variant marked as filtered, but no failed filters listed in %s.".format(v))
+            vcb.filters(failedFilters.mkString(";"))
+          }
+        }).getOrElse({
+          throw new IllegalArgumentException("Filters were applied but filters passed is null in %s.".format(v))
+        })
+      }).getOrElse(vcb.unfiltered)
+  }
+
+  private val variantExtractFns: Iterable[(Variant, VariantContextBuilder) => VariantContextBuilder] = Iterable(
+    extractNames(_, _),
+    extractFilters(_, _)
+  )
+
+  // htsjdk --> variant annotation format functions
+
+  private[converters] def formatAncestralAllele(
+    vc: HtsjdkVariantContext,
+    vab: VariantAnnotation.Builder,
+    v: Variant,
+    index: Int): VariantAnnotation.Builder = {
+
+    Option(vc.getAttributeAsString("AA", null))
+      .fold(vab)(vab.setAncestralAllele(_))
+  }
+
+  private[converters] def formatDbSnp(
+    vc: HtsjdkVariantContext,
+    vab: VariantAnnotation.Builder,
+    v: Variant,
+    index: Int): VariantAnnotation.Builder = {
+
+    Option(vc.getAttribute("DB").asInstanceOf[java.lang.Boolean])
```

Comment:
Is this how the `getAttribute` function works on `VariantContext`? I remember having the sad finding that it didn't work like this on `Genotype`.